src/tests/end-to-end/utils/utils.h

```diff
@@ -8,13 +8,64 @@
 
 using namespace Antares::Solver;
 using namespace Antares::Solver::Simulation;
+using namespace Antares::Data::ScenarioBuilder;
 
 
 void initializeStudy(Study::Ptr study);
-void setNumberMCyears(Study::Ptr study, unsigned int nbYears);
 void configureLinkCapacities(AreaLink* link);
+
+
+template<class MatrixType>
+class TimeSeriesConfig
+{
+public:
+    TimeSeriesConfig(MatrixType& matrix) : ts_(&matrix) {}
+    TimeSeriesConfig& setNumberColumns(unsigned int nbColumns);
+    TimeSeriesConfig& fillColumnWith(unsigned int column, double value);
+private:
+    MatrixType* ts_;
```

Comment:
Maybe it would be nice to have a common `IMatrix` interface, so we don't need to templatize everything.

Something like
```cpp
class IMatrix {
    virtual std::size_t width() const;
    virtual std::size_t height() const;
    virtual void resize(std::size_t width, std::size_t height);
};

template<class T, class U>
class Matrix : public IMatrix {
 // ...
};
```

Reply:
If I understand you properly, we would inject the Matrix in the **TimeSeriesConfigurer** at construction, I am right ?

Reply:
You'd inject it's `IMatrix` part through [object slicing](https://en.wikipedia.org/wiki/Object_slicing)